wireless/bluetooth/bt_netdev.c

```diff
@@ -1194,7 +1194,7 @@ int bt_netdev_register(FAR struct bt_driver_s *btdev)
 
   /* Get the interface structure associated with this interface number. */
 
-  priv = (FAR struct btnet_driver_s *)
+  btdev->priv = priv = (FAR struct btnet_driver_s *)
```

Comment:
but this field is used by driver self not the bluetooth stack?

Reply:
The driver should not use this field.  The file bt_driver.h has this comment:
/* Filled by register function, shouldn't be touched by bt_driver_s */
I would like to create a bt_netdev_unregister(FAR struct bt_driver_s *btdev) function that does this:
{
 FAR struct btnet_driver_s *priv;
priv = (FAR struct btnet_driver_s *)btdev->priv; <-------- 
netdev_unregister(&priv->bd_dev.r_dev);
}

Reply:
but many drivers already use this field to save it's context:(.

Reply:
This is clearly a private field that is reserved for the stack. My bt_netdev_unregister example above shows why the stack **needs** this field. The comment supports this use case.  The correct way to use this "interface" is to create a "subclass".

struct my_driver {
  bt_driver_s btdev;
 void *priv;
}

To answer your question directly: No, we should not change the comment in bt_driver_s (bt_driver.h). I have given my reasons above. Any drivers abusing the bt_driver_s->priv field needs to be corrected. For other structs, the community can decide.

Reply:
If so, this patch needs to fix the drivers which use this field to save self context before it can be merged.

Reply:
Thanks for your input. Right now I will have to fork and finish my USB bluetooth project in a private branch. I need to make sure this stack will work with removable USB bluetooth modules before I commit any more time to this PR. If I feel the need to upstream my changes I will see if I can fix those drivers. 
Thanks.

Reply:
thanks